name: RH Workflow Test

on:
  push:
    branches: 
    - unittest-rh

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: Echo Test
      run: echo 'Hello World'
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 2.2.108
    - name: Install dependencies
      run: dotnet restore
      working-directory: 'apis/poi/'
    - name: Build
      run: dotnet build --configuration Release --no-restore
      working-directory: 'apis/poi/'
    - name: .Net Test
      working-directory: 'apis/poi/'
      run: dotnet test --no-restore --verbosity normal
    - name: Check Failure
      if: ${{ failure() }}
      run: |
          curl --request POST \
            --url https://api.github.com/repos/${{ github.repository }}/issues \
            --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
            --header 'content-type: application/json' \
            --data '{
              "title": "Issue created by RH workflow test upon unit test failure: ${{ github.run_id }}",
              "body": "This issue was automatically created by the GitHub Action workflow **${{ github.workflow }}**."
              }'
